src/utils/utils.rs

```diff
@@ -64,7 +64,7 @@ pub(crate) fn append_file(name: &'static str, path: &Path, line: &str) -> Result
 
 pub(crate) fn write_line(
     name: &'static str,
-    file: &mut File,
+    file: &mut dyn Write,
```

Comment:
I've noticed in the past that the vtable indirection can cost some performance for I/O, and given that there's only one call site it probably couldn't hurt to use `impl Write` instead of `dyn Write` here.

(IMO we might also instead just inline `utils::write_line()` given that it only has one caller, which would avoid needing to specify the type -- but that should probably be in a separate commit.)

Reply:
Ok.